types/metadataV14_portable.go

```diff
@@ -77,19 +77,21 @@ func (d *Si0TypeDefPrimitive) Decode(decoder scale.Decoder) error {
 
 //------------------v1-----------
 
-type Si1LookupTypeID big.Int
+type Si1LookupTypeID struct {
+	UCompact
```

Comment:
I realized that we already have `type Si0LookupTypeID UCompact`, ie., a type alias. If you don't feel strong about defining it as a struct, we would benefit from being consistent and have `type Si1LookupTypeID UCompact` instead.

Reply:
Ok, But the inheritance of go language is a bit strange, If you use `type Si1LookupTypeID UCompact`, you have to rewrite the `encode` and `decode` methods belonging to `Si0LookupTypeID `. I use the structure method to make `Si0LookupTypeID ` inherit the methods of `UCompact`. 
Or you have other better ideas ?🤔

Reply:
I am new to Go, didn't know that. Sounds good 👍

Reply:
😂 I switched from Java to Go. It’s strange to me that the inheritance of Go. I once had the same idea as you.